### PR TITLE
AnalyzerWaveformTest refactoring as a foundation for more tests

### PR DESCRIPTION
**Pull request by daschuer**
The next step is to add a test the compares the resulting waveform.

### CODE REVIEW

File: src/test/analyserwaveformtest.cpp [hunk 1]
```diff
@@ -8,79 +8,68 @@
...
+        m_canaryBigBuf = new CSAMPLE[kBigBufSize + 2 * kCanarySize];
+        for (std::size_t i = 0; i < kCanarySize; i++) {
+            m_canaryBigBuf[i] = kCanaryFloat;
+        }
+        for (std::size_t i = kCanarySize; i < kCanarySize + kBigBufSize; i++) {
+            m_canaryBigBuf[i] = kMagicFloat;
+        }
+        for (std::size_t i = kCanarySize + kBigBufSize; i < 2 * kCanarySize + kBigBufSize; i++) {
+            m_canaryBigBuf[i] = kCanaryFloat;
+        }
```
Comment (Swiftb0y) [inline src/test/analyserwaveformtest.cpp:45]: couldn't these be `memset`s?

**Review by Swiftb0y — COMMENTED** [review]
The canary test is definitely a little weird and would still break if the analyzer would overrun its buffer significantly. Why do we not bounds check instead?

File: src/test/analyserwaveformtest.cpp (hunk 1 above)
Comment (daschuer) [reply src/test/analyserwaveformtest.cpp:45]: memset() works with char arrays only. We can do a lot more refactoring here, but I like to stop at this point.

**Comment by daschuer** [comment]
> The canary test is definitely a little weird and would still break if the analyzer would overrun its buffer significantly. Why do we not bounds check instead?

The whole test is kind of pointless, because the buffer is passed to the analyzer as const. So there must be went a lot of things wrong to make this test fail. Maybe it was relevant for an early version of the analyzer? We may consider to remove it.

File: src/test/analyserwaveformtest.cpp (hunk 1 above)
Comment (Swiftb0y) [reply src/test/analyserwaveformtest.cpp:45]: Right, I forgot about that.

**Review by Swiftb0y — APPROVED** [review]

**Pull request merged**